modules/oe_theme_helper/tests/src/Functional/RetinaScaleEffectTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_helper\Functional;
+
+use Drupal\FunctionalTests\Image\ToolkitTestBase;
+
+/**
+ * Tests that the Retina Scale effect upscales images appropriately.
+ *
+ * @group image
+ */
+class RetinaScaleEffectTest extends ToolkitTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['image', 'oe_theme_helper'];
+
+  /**
+   * The image effect manager.
+   *
+   * @var \Drupal\image\ImageEffectManager
+   */
+  protected $manager;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->manager = $this->container->get('plugin.manager.image.effect');
+  }
+
+  /**
+   * Test the retina_image_scale_effect() function.
+   */
+  public function testRetinaScaleEffect(): void {
+    $this->assertImageEffect('retina_image_scale', [
+      // Set the desired width to be much higher than the image width.
+      'width' => $this->image->getWidth() * 10,
+    ]);
+    $this->assertToolkitOperationsCalled(['scale']);
+
+    // Check the parameters.
+    $calls = $this->imageTestGetAllCalls();
+    $this->assertEqual($calls['scale'][0][0], $this->image->getWidth() * 2, 'Width is double the original size.');
+  }
+
+  /**
+   * Test the image_scale_effect() function using a multiplier of 3.
+   */
+  public function testTripleMultiplierRetinaScaleEffect(): void {
+    $this->assertImageEffect('retina_image_scale', [
+      // Set the desired width to be much higher than the image width.
+      'width' => $this->image->getWidth() * 10,
+      // Set a multiplier of three for the test.
+      'multiplier' => 3,
+    ]);
+    $this->assertToolkitOperationsCalled(['scale']);
+
+    // Check the parameters.
+    $calls = $this->imageTestGetAllCalls();
+    $this->assertEqual($calls['scale'][0][0], $this->image->getWidth() * 3, 'Width is triple the original size.');
+  }
+
+  /**
+   * Test the image_scale_effect() function using a big enough image.
+   */
+  public function testScaleEffect(): void {
+    $this->assertImageEffect('image_scale', [
+      // Set the image width to be smaller than the image width.
```

Comment:
Comment sounds weird. Maybe keep it in line with the same comment in the previous methods?